deepchem/data/datasets.py

```diff
@@ -96,6 +96,8 @@ def pad_batch(batch_size, X_b, y_b, w_b, ids_b):
 
     # Fill in batch arrays
     start = 0
+    # Only the first set of copy will be counted in training loss
+    w_out[start:start + num_samples] = w_b[:]
```

Comment:
the real question is why can't we use this technique for X_out and y_out as well